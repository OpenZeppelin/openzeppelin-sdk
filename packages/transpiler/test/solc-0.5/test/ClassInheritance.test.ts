import { shouldTranspileToValidContract } from './LocalConract.behaviour';

describe(`ClassInheritance contract`, (): void => {
  shouldTranspileToValidContract('CIB', {
    CIB: {
      path: 'ClassInheritance',
      fileName: 'ClassInheritance',
      contracts: ['CIB', 'CIA'],
    },
    CIA: {
      path: 'ClassInheritance',
      fileName: 'ClassInheritance',
      contracts: ['CIA'],
    },
  });
});
