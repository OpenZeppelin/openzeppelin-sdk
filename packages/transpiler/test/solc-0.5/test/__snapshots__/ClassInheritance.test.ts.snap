// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClassInheritance contract is converted to a valid CIBUpgradable contract 1`] = `
"pragma solidity ^0.5.0;
import \\"@openzeppelin/upgrades/contracts/Initializable.sol\\";

contract CIAUpgradable is Initializable {
    function initialize(uint bar) external initializer {
        __init(true, bar);
    }

    function __init(bool callChain, uint bar) internal {
        
        
        
        foo = bar;
        emit log(\\"SIA\\");
    
    }

    uint256 public foo;
    event log(string);
    
}

contract CIBUpgradable is Initializable, CIAUpgradable {
    function initialize() external initializer {
        __init(true);
    }

    function __init(bool callChain) internal {
        if(callChain) {
            CIAUpgradable.__init(false, 324);
        }
        
        val = 123;
        
    }

    uint256 public val ;
}"
`;
